libbeat/template/template.go

```diff
@@ -51,7 +48,7 @@ type Template struct {
 }
 
 // New creates a new template instance
-func New(beatVersion string, beatName string, esVersion string, config TemplateConfig) (*Template, error) {
+func NewTemplate(beatVersion string, beatName string, esVersion string, config TemplateConfig) (*Template, error) {
```

Comment:
Why rename?

Reply:
It is more aligned with the general libbeat code, and since the `template` code needs to be touched anyways, it is a good opportunity to adapt this.

Reply:
I would keep the same name as before, we are in the template package so no need to repeat it in the constructor.

template.New instead of template.NewTemplate.